generator/src/main/java/com/scottlogic/deg/generator/decisiontree/MaxStringLengthInjectingDecisionTreeFactory.java

```diff
@@ -0,0 +1,64 @@
+package com.scottlogic.deg.generator.decisiontree;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.constraints.atomic.IsStringShorterThanConstraint;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.inputs.RuleInformation;
+import com.scottlogic.deg.schemas.v0_1.RuleDTO;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Decorator over a DecisionTreeFactory to inject a &lt;shorterThan X&gt; constraint at the root node for every field
+ */
+public class MaxStringLengthInjectingDecisionTreeFactory implements DecisionTreeFactory{
+    private final int maxLength;
+    private final DecisionTreeFactory underlyingFactory;
+
+    @Inject
+    public MaxStringLengthInjectingDecisionTreeFactory(ProfileDecisionTreeFactory underlyingFactory) {
+        this(underlyingFactory, GenerationConfig.Constants.MAX_STRING_LENGTH.intValue());
+    }
+
+    public MaxStringLengthInjectingDecisionTreeFactory(DecisionTreeFactory underlyingFactory, int maxLength) {
+        this.underlyingFactory = underlyingFactory;
+        this.maxLength = maxLength;
+    }
+
+    @Override
+    public DecisionTreeCollection analyse(Profile profile) {
+        DecisionTreeCollection collection = underlyingFactory.analyse(profile);
+        return new DecisionTreeCollection(
+            collection.getFields(),
+            collection.getDecisionTrees()
+                .stream()
+                .map(this::injectConstraints)
+                .collect(Collectors.toList())
+        );
+    }
+
+    private DecisionTree injectConstraints(DecisionTree tree) {
```

Comment:
Why would we do this here, a generation default would be much less messy to just use where it's needed in the generator

Reply:
From my perspective: 'cos it's better for things like equivalence class derivation if all restrictions are represented in the logic tree.